### ISSUE TITLE
Support catalog description string

### ISSUE THREAD

**Issue by mosabua** [trinodb/trino#11150]

**Comment by abrisan** [comment]
I’ve been wanting to contribute to core trino for some time and this looks like a good first contribution! Any pointers on where I could start?

**Comment by mosabua** [comment]
Sorry .. not really @abrisan  .. you would have to dig into the source to find out more.

**Comment by abrisan** [comment]
Sure, no problem. Do you have any sort of timeline you need? I was thinking of taking a stab at it this weekend :)

**Comment by dain** [comment]
This will likely be an annoying project to start with.   We need to have clarity on where the information would be show.  For example, comments are not show in the other show commands, so that doesn't seem like a good place.  For tables, I believe they show up in SHOW CREATE TABLE, but there is no command like that for catalogs (because you can create one).  You could add it to the system metadata table `select * from system.metadata.catalogs`.  I'm not sure there is a way get this information from JDBC or other clients.... anyway, my point is, I this needs discussion before someone works on it.

**Comment by mosabua** [comment]
Yeah .. agreed. This definitely goes into many parts of the system and is not easy to implement.

**Comment by mosabua** [comment]
Also important note on use case:

Imagine a cluster has dozens of catalogs with Hive, Iceberg and various others. If a user needs to find out more information about the catalogs it should be possible to do that. E.g. an important aspect can be what connector it used to know what table properties are supported for CTAS query or materialized views. A general requirement often is also what the scope of the catalog is and often you have to come up with naming conventions and so on. It would be nice if there were an easier and more comprehensive way to manage and expose metadata about catalogs.

**Comment by raj-manvar** [comment]
+1 to this feature!
